packages/blade/src/components/Chip/Chip.stories.tsx

```diff
@@ -0,0 +1,564 @@
+            </Text>
+
+            <ChipGroupComponent
+              size={size as ChipGroupProps['size']}
+              defaultValue="Proprietorship"
+              {...args}
+            >
+              {chipValues.map((chipValue: string) => (
+                <ChipComponent key={chipValue} value={chipValue}>
+                  {chipValue}
+                </ChipComponent>
+              ))}
+            </ChipGroupComponent>
+          </Box>
+          <Box marginBottom="spacing.3">
+            <Heading size="small">with Icon</Heading>
+            <Text marginBottom="spacing.3" marginTop="spacing.3" size="medium">
+              What other capabilities are you looking for?
+            </Text>
+
+            <ChipGroupComponent
+              accessibilityLabel="What other capabilities are you looking for?"
+              defaultValue="Proprietorship"
+              size={size as ChipGroupProps['size']}
+            >
+              <ChipComponent value="payment-links" icon={PaymentLinksIcon}>
+                Automated Payment Links
+              </ChipComponent>
+              <ChipComponent value="wallet" icon={SmartphoneIcon}>
+                Wallet on My App
+              </ChipComponent>
+              <ChipComponent value="offers" icon={OffersIcon}>
+                Offer discounts, Pay Later & EMI options
+              </ChipComponent>
+            </ChipGroupComponent>
+          </Box>
+        </Box>
+      ))}
+    </Box>
+  );
+};
+
+export const AllChipSizes = AllChipSizesTemplate.bind({});
+AllChipSizes.storyName = 'All Sizes';
+
+const ChipIntentsTemplate: ComponentStory<typeof ChipGroupComponent> = (args) => {
+  return (
+    <Box display="flex" flexDirection="column">
+      <Text size="large" weight="bold" marginBottom="spacing.3">
+        Is the result helpful?
+      </Text>
+
+      <ChipGroupComponent defaultValue="yes" {...args}>
+        <ChipComponent intent="positive" value="yes" icon={ThumbsUpIcon}>
+          Yes
+        </ChipComponent>
+        <ChipComponent intent="negative" value="no" icon={ThumbsDownIcon}>
+          No
+        </ChipComponent>
+      </ChipGroupComponent>
+    </Box>
+  );
+};
+
+export const ChipWithIntent = ChipIntentsTemplate.bind({});
+ChipWithIntent.storyName = 'With Intent';
+ChipWithIntent.args = {
+  accessibilityLabel: 'Is the result helpful?',
+};
+
+const TextTransformationTemplate: ComponentStory<typeof ChipGroupComponent> = ({
+  children,
+  ...args
+}) => {
+  const chipValues = ['Proprietorship', 'Public', 'Small Business'];
+  return (
+    <Box>
+      <Text marginBottom="spacing.3" marginTop="spacing.3">
+        Select Business type:
+      </Text>
+
+      <ChipGroupComponent defaultValue="Proprietorship" {...args}>
+        {chipValues.map((chipValue: string) => (
+          <ChipComponent key={chipValue} value={chipValue}>
+            {chipValue.toUpperCase()}
+          </ChipComponent>
+        ))}
+      </ChipGroupComponent>
+    </Box>
```

Comment:
lets put a code component here with an example stating that you can pass `value.toUpperCase()` to the children to make it uppercase because on storybook they parse the JS so they can't really identify how to make this work

<img width="1075" alt="Screenshot 2023-08-28 at 1 15 08 PM" src="https://github.com/razorpay/blade/assets/11384858/b64537c9-3701-4c28-bae7-5689e2dbe11e">

Reply:
added